model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_advection_in_vertical_momentum_equation.py

```diff
@@ -48,40 +48,49 @@ def _compute_advective_vertical_wind_tendency_and_apply_diffusion(
     nlev: gtx.int32,
     nrdmax: gtx.int32,
 ) -> fa.CellKField[ta.vpfloat]:
-    k = broadcast(k, (dims.CellDim, dims.KDim))
-
-    vertical_wind_advective_tendency = where(
-        (cell_lower_bound <= cell < cell_upper_bound) & (1 <= k),
-        _compute_advective_vertical_wind_tendency(
-            contravariant_corrected_w_at_cells_on_half_levels, w, coeff1_dwdz, coeff2_dwdz
+    vertical_wind_advective_tendency = concat_where(
+        1 <= dims.KDim,
+        where(
+            cell_lower_bound <= cell < cell_upper_bound,
+            _compute_advective_vertical_wind_tendency(
+                contravariant_corrected_w_at_cells_on_half_levels, w, coeff1_dwdz, coeff2_dwdz
+            ),
+            vertical_wind_advective_tendency,
         ),
         vertical_wind_advective_tendency,
     )
-    vertical_wind_advective_tendency = where(
-        (cell_lower_bound <= cell < cell_upper_bound) & (1 <= k),
-        _add_interpolated_horizontal_advection_of_w(
-            e_bln_c_s,
-            horizontal_advection_of_w_at_edges_on_half_levels,
+    vertical_wind_advective_tendency = concat_where(
+        1 <= dims.KDim,
+        where(
+            cell_lower_bound <= cell < cell_upper_bound,
```

Comment:
we missed this one, I propose to go ahead and fix it in a next iteration (later today/this week) to unblock other work

Reply:
we just need to delete the index field constructors, then we should have caught all